src/ucp/proto/proto_am.inl

```diff
@@ -26,14 +26,21 @@ static UCS_F_ALWAYS_INLINE ucs_status_t
 ucp_do_am_bcopy_single(uct_pending_req_t *self, uint8_t am_id,
                        uct_pack_callback_t pack_cb)
 {
-    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
-    ucp_ep_t *ep       = req->send.ep;
+    ucp_request_t *req   = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_ep_t *ep         = req->send.ep;
+    ucp_dt_state_t state = req->send.state.dt;
     ssize_t packed_len;
 
     req->send.lane = ucp_ep_get_am_lane(ep);
     packed_len     = uct_ep_am_bcopy(ep->uct_eps[req->send.lane], am_id, pack_cb,
                                      req, 0);
-    if (packed_len < 0) {
+    ucs_assertv((packed_len < 0) ||
+                (packed_len <= ucp_ep_get_max_bcopy(ep, req->send.lane)),
```

Comment:
this assert still looks excessive/confusing to me. 
maybe leave just `packed_len <= ucp_ep_get_max_bcopy(ep, req->send.lane)` check and move it to line 46 before return?

Reply:
1. what if `packed_len == -2`? i.e. `packed_len == UCS_ERR_NO_RESOURCE` when it will be converted to `size_t`, it will be `SIZE_MAX - 1`, so the assert reaises SIGABRT signal
2. what if `packed_len == 8`, and we add casting `packed_len <= (ssize_t)ucp_ep_get_max_bcopy(ep, req->send.lane)`, it still would not work if `UCX_RNDV_THRESH = UCX_ZCOPY_THRESH = inf`, (i.e. they are `== -1`)

Reply:
you may omit checking for `<0` in the assert, because you have this check right under this assert
then if you can move this assert below `if (packed_len < 0)` and assume that packed_len is positive

i did not get the seconds example

Reply:
> you may omit checking for `<0` in the assert, because you have this check right under this assert
> then if you can move this assert below `if (packed_len < 0)` and assume that packed_len is positive

ok, it makes sense to me - done